keras_hub/src/models/flux/flux_model.py

```diff
@@ -0,0 +1,234 @@
+        image_ids: KerasTensor. Image ID input tensor of shape (N, L, D) corresponding
+                to the image sequences.
+        text: KerasTensor. Text input tensor of shape (N, L, D).
+        text_ids: KerasTensor. Text ID input tensor of shape (N, L, D) corresponding
+            to the text sequences.
+        timesteps: KerasTensor. Timestep tensor used to compute positional embeddings.
+        y: KerasTensor. Additional vector input, such as target values.
+        guidance: KerasTensor, optional. Guidance input tensor used
+            in guidance-embedded models.
+    Raises:
+        ValueError: If `hidden_size` is not divisible by `num_heads`, or if `sum(axes_dim)` is not equal to the
+                    positional embedding dimension.
+    """
+
+    def __init__(
+        self,
+        input_channels,
+        hidden_size,
+        mlp_ratio,
+        num_heads,
+        depth,
+        depth_single_blocks,
+        axes_dim,
+        theta,
+        use_bias,
+        guidance_embed=False,
+        # These will be inferred from the CLIP/T5 encoders later
+        image_shape=(None, 768, 3072),
+        text_shape=(None, 768, 3072),
+        image_ids_shape=(None, 768, 3072),
+        text_ids_shape=(None, 768, 3072),
+        y_shape=(None, 128),
+        timestep_shape=(256,),
+        guidance_shape=(256,),
+        **kwargs,
+    ):
+
+        # === Layers ===
+        self.positional_embedder = EmbedND(theta=theta, axes_dim=axes_dim)
+        self.image_input_embedder = keras.layers.Dense(
+            hidden_size, use_bias=True
+        )
+        self.time_input_embedder = MLPEmbedder(hidden_dim=hidden_size)
+        self.vector_embedder = MLPEmbedder(hidden_dim=hidden_size)
+        self.guidance_input_embedder = (
+            MLPEmbedder(hidden_dim=hidden_size)
+            if guidance_embed
+            else keras.layers.Identity()
+        )
+        self.text_input_embedder = keras.layers.Dense(hidden_size)
+
+        self.double_blocks = [
+            DoubleStreamBlock(
+                hidden_size,
+                num_heads,
+                mlp_ratio=mlp_ratio,
+                use_bias=use_bias,
+            )
+            for _ in range(depth)
+        ]
+
+        self.single_blocks = [
+            SingleStreamBlock(hidden_size, num_heads, mlp_ratio=mlp_ratio)
+            for _ in range(depth_single_blocks)
+        ]
+
+        self.final_layer = LastLayer(hidden_size, 1, input_channels)
+        self.timestep_embedding = TimestepEmbedding()
+        self.guidance_embed = guidance_embed
+
+        # === Functional Model ===
+        image_input = keras.Input(shape=image_shape, name="image")
+        image_ids = keras.Input(shape=image_ids_shape, name="image_ids")
+        text_input = keras.Input(shape=text_shape, name="text")
+        text_ids = keras.Input(shape=text_ids_shape, name="text_ids")
+        y = keras.Input(shape=y_shape, name="y")
+        timesteps_input = keras.Input(
```

Comment:
@divyashreepathihalli this is a non-standard approach. The timesteps and guidance seem to have to be flat (unbatched) for the operations to work as they do in the original implementation. The same goes for the original impl - they're unbatched.

This is currently a hacky way to do this. Is there a preferred/cleaner alternative?